src/IssueViz.Security/IssueVizsStore.cs

```diff
@@ -0,0 +1,101 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using SonarLint.VisualStudio.Core.Helpers;
+using SonarLint.VisualStudio.IssueVisualization.Editor.LocationTagging;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Security
+{
+    internal interface IIssueVizsStore : IIssueLocationStore, IDisposable
+    {
+        ReadOnlyObservableCollection<IAnalysisIssueVisualization> GetAll();
+    }
+
+    internal sealed class IssueVizsStore : IIssueVizsStore
```

Comment:
This is a repeating identical logic between hotspots and taint, and we already had a bug in it (the null path, which will be relevant for taint as well). I created a common Store for both of them to use.
Take a look at how the usage will be like:
https://github.com/SonarSource/sonarlint-visualstudio/pull/2006/files#diff-1c9f171be0a53939909d03bedba2f81a5e7782a0a80c2c0b78dc913d74d360b7